### PR TITLE
Add HtmlProofer to validate links as well as fix current issues

### PR DESCRIPTION
**Pull request by stevesloka**
Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by stevesloka** [comment]
@SDBrett

**Comment by stevesloka** [comment]
I didn't realize but this might be similar to #1848

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ vet: | test
 	go vet $(MODULE)/...
 
 check: ## Run tests and CI checks
-check: test test-race vet gofmt staticcheck misspell unconvert unparam ineffassign yamllint check-stale
+check: test test-race vet gofmt staticcheck misspell unconvert unparam ineffassign yamllint check-stale site-check
```
Comment (jpeach) [inline Makefile]: My workflow is normally `make check`, but the Docker install takes a long time. I can switch to `make test` but not sure whether we want `check` to take longer by default.

File: Makefile [hunk 2]
```diff
@@ -218,6 +218,10 @@ site-devel: ## Launch the website in a Docker container
 	docker run --publish $(JEKYLL_PORT):$(JEKYLL_PORT) -v $$(pwd)/site:/site -it $(JEKYLL_IMAGE) \
 		bash -c "cd /site && bundle install && bundle exec jekyll serve --host 0.0.0.0 --port $(JEKYLL_PORT) --livereload"
 
+site-check: ## Launch the website in a Docker container
```
Comment (jpeach) [inline Makefile]: Can you please update the doc comment?

File: site/_config.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ footer_social_links:
     url: https://kubernetes.slack.com/messages/contour
   RSS:
     fa_icon: fa fa-rss
-    url: feed.xml
+    url: http://projectcontour.io/feed.xml
```
Comment (jpeach) [inline site/_config.yml]: HTTPS?

File: site/_config.yml [hunk 4]
```diff
@@ -46,6 +46,7 @@ defaults:
       layout: "default"
 
 repository: projectcontour/contour
+github: [metadata]
```
Comment (jpeach) [inline site/_config.yml]: What does this do?

File: site/_posts/2019-07-23-contour-v014.md [hunk 5]
```diff
@@ -36,7 +36,7 @@ Until Contour release v0.14, the deployment model placed Contour and Envoy in th
...
-The Contour repo includes step-by-step examples of how to generate certificates from a command line if you want to [generate them by hand](https://github.com/projectcontour/contour/blob/master/docs/grpc-tls-howto.md#generating-example-grpc-tls-certificates); the example/ds-hostnet-split [example](https://github.com/projectcontour/contour/tree/master/examples/ds-hostnet-split) includes a [job](https://github.com/projectcontour/contour/blob/master/examples/ds-hostnet-split/02-job-certgen.yaml) which automatically generate the certificates, or you could provide your own based on your IT security requirements.
+The Contour repo includes step-by-step examples of how to generate certificates from a command line if you want to [generate them by hand](https://projectcontour.io/guides/grpc-tls-howto#generating-example-grpc-tls-certificates); the example/contour [example](https://github.com/projectcontour/contour/blob/master/examples/contour) includes a [job](https://github.com/projectcontour/contour/blob/master/examples/contour/02-job-certgen.yaml) which automatically generate the certificates, or you could provide your own based on your IT security requirements.
```
Comment (jpeach) [inline site/_posts/2019-07-23-contour-v014.md]: Can you use `site.github.repository_url` for these links?

File: site/_resources/envoy.md [hunk 6]
```diff
@@ -23,7 +23,7 @@ This page describes the compatibility matrix of Contour and Envoy versions.
 ## Envoy extensions
 
 Contour requires the following extensions.
-If you are using the image recommended in our [example deployment](https://github.com/projectcontour/contour/blog/{{ site.github.latest_release.tag_name }}/examples/contour) no action is required.
+If you are using the image recommended in our [example deployment](https://github.com/projectcontour/contour/tree/{{ site.github.latest_release.tag_name }}/examples/contour) no action is required.
```
Comment (jpeach) [inline site/_resources/envoy.md]: Can you use `site.github.repository_url` here?

File: site/_guides/tls.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
+title: TLS support
+layout: page
+---
+
+# TLS support
+
+Contour 0.3 adds support for HTTPS (TLS/SSL) ingress by integrating Envoy's SNI support.
```
Comment (jpeach) [inline site/_guides/tls.md]: Isn't 0.3 ancient history? How about we just say that Contour supports TLS for Ingress and HTTPProxy?

**Comment by SDBrett** [comment]
Over all it LGTM.

**Pull request merged**